### ISSUE TITLE
color contrast accessibility myths

### ISSUE THREAD

**Issue by nirjan-dev** [nirjan-dev/learning-to-code#41]
[link to the article](https://uxmovement.com/buttons/the-myths-of-color-contrast-accessibility/)
Myth 1: The WCAG requirements are always optimal.
he WCAG contrast ratios don’t always account for the high luminance contrast of white text. White is pure luminance with no hue or saturation, which is the strongest form for contrast. Therefore, it makes sense why the button with white text is easier to read (source).

The reason the contrast ratios failed with the white text is that it has high luminance and is on a background with high luminance. Light text on a light background is rendered low contrast computationally. Your design is supposed to satisfy what people see, not computational algorithms. It’s why the designer’s eye should always play a part in the equation.

The WCAG are guidelines to help designers choose the right color contrasts. The adage, “The map is not the territory,” applies here. Don’t confuse models of reality with reality itself.

Myth 2: Text needs to meet the AAA requirement, or it’s inaccessible.
A vision loss of 20/80 is rare among the general population and mostly affects the elderly suffering from age-related eye diseases. A study found that most low vision is related to aging (source). If the majority of your user base is 70 or older, meeting the AAA requirement is beneficial. The standard is 70 or older because visual acuity starts to decline among users with healthy eyes at that age (source).

Meeting the AA requirement is sufficient for the majority of users. The AA requirement constitutes a contrast ratio of 4.5:1 to compensate for the loss of contrast sensitivity by users with a 20/40 vision loss. A study found the “majority of persons maintain at least fair acuity (20/40 or better) into their 80’s” (source). This finding means that meeting the AA requirement will make your text accessible to the majority of users.

Myth 3: Interface components have the same contrast ratio standard as text.
Many make the mistake of holding interface components to the same contrast ratio standard as text when they are different. Interface components have a contrast ratio of 3:1, while text is 4.5:1. Text requires a higher contrast because users need to read it. Interface components don’t require reading and have a lower standard
Many nuances affect text contrast, such as font size and weight. Large text sizes (18 pt) and text with heavier font weights (14 pt bold) require lower contrast ratios (source). Not only that, but certain interface components are exempt from the requirement. Before you hold an interface component or text to a contrast ratio standard, make sure you’re applying it correctly in the right situations.

Myth 4: Gray text and buttons are inaccessible and look disabled.
Another common myth is that gray text is inaccessible. Many assume users can’t read gray text because it looks low contrast. Sometimes this may be true, but other times it’s a false assumption. For example, the button below has gray text and some would assume it’s inaccessible. However, running it through a contrast checker shows that it’s not only AA compliant, but the ratio is well above the standard.

The other myth you might hear is that a gray button is inaccessible because it doesn’t meet the contrast ratio standard. It turns out the success criteria for buttons doesn’t require a visual boundary indicating the hit area. If a button with text has a border, there is no contrast requirement beyond the text contrast (source). Therefore, the gray button that most would assume is inaccessible passes the contrast requirement.

This success criteria also means that icons next to buttons don’t have a contrast requirement as long as the text label meets the 4.5:1 contrast ratio. However, if an icon is without a text label, the 3:1 contrast ratio requirement applies to the icon.

There’s also the myth that gray buttons look disabled, which is often parroted by biased observers who don’t understand the proper signifier for inactive components. Disabled buttons are signified by the lack of contrast to the text label. When a button is hard to read, users don’t bother with it, which is the intent of a disabled button. Not to mention, the contrast requirement does not apply to inactive components.

Myth 5: Color blind users can’t tell the difference between contrasting colors.

If you’re using competing color hues to differentiate states, you need another visual cue besides color. But if you’re only using color contrast to differentiate states, it’s likely accessible to color blind users.

There are various types of color blindness, but the ones you should focus on the most are red-green deficiencies. Red-green color blindness affects more than 99% of all color blind people (source). There are several color blindness simulators you can choose from on Chrome extensions, such as Colorblindly.

Myth 6: Using a color cue alone isn’t sufficient in conveying information.
![image](https://user-images.githubusercontent.com/29270715/68275522-05c24900-0094-11ea-9dc8-d097191cca4c.png)
![image](https://user-images.githubusercontent.com/29270715/68275527-0955d000-0094-11ea-855f-5e489cb4ccc1.png)
![image](https://user-images.githubusercontent.com/29270715/68275532-0c50c080-0094-11ea-94ef-9bc68f077d28.png)

Myth 7: An accessible design meets the needs of every user on the planet.
Every designer would love to meet the needs of every user on the planet. As much as designers strive for this, it’s impossible to achieve. Even if you followed the WCAG to the tee, there’s always some user who will find your design uneasy on their eyes.

Instead of striving for an ideal based on an unachievable fantasy, strive for one based on an achievable reality. The reality is that an accessible design can’t meet the needs of every user, but it can meet the needs of as many users as possible.

Understanding this truth means accepting the fact that a minority of users won’t have as good of an experience as the majority. But luckily for the minority, there are assistive technologies with high contrast modes available to them. Designers who truly understand accessibility will strive to achieve the realistic ideal, not the fantasy one.

The Nuances of Color Contrast Accessibility
Accessibility should always be a priority when designing for users. The WCAG guidelines are an effective tool to help you achieve an accessible design of the highest standard. These myths are not caused by the WCAG guidelines. They are caused by people who misinterpret, misrepresent, and misuse the guidelines. It’s time to put these myths to rest.

Understanding the nuances of color contrast accessibility will help you meet the WCAG standards accurately. When others project color contrast accessibility myths onto your design, you can correct them. You’ll stay true to visual simplicity and aesthetics while balancing it with accessibility at the same time. The result is an inclusive interface that satisfies almost everyone.
